app/models/article.rb

```diff
@@ -399,8 +399,15 @@ def update_notifications
   def before_destroy_actions
     bust_cache
     remove_algolia_index
-    user.cache_bust_all_articles
-    Articles::ResaveJob.perform_later(organization.article_ids - [id]) if organization
+    article_ids = user.article_ids.dup
+    if organization
+      organization.touch(:last_article_at)
+      article_ids.concat organization.article_ids
+    end
+    # perform busting cache in chunks in case there're a lot of articles
+    (article_ids.uniq.sort - [id]).each_slice(10) do |ids|
```

Comment:
Sorting here is just to be able to test, that the job is enqueued with the correct arguments (`spec/models/article_destroy_spec.rb:24`). The order of article_ids is not important. Sets are not allowed as job arguments.
If anyone has a better idea to remove sorting and still test, please, share.